### ISSUE TITLE
dataset

### ISSUE THREAD

**Issue by 3120932798** [xXAI-botXx/torch-mask-rcnn-instance-segmentation#1]
你好 我您的这个数据集可以转成coco格式的吗

**Comment by M-106** [comment]
Thank you for your question. This repository only contains the code for Mask-RCNN and does not contain a dataset, so I understand your question like: "Does this model support datasets in COCO format?"
Answer is: not right now and it is not planned. The data format for this model is decribed in the [data preparation section](https://github.com/xXAI-botXx/torch-mask-rcnn-instance-segmentation?tab=readme-ov-file#data-preparation).

If you need a dataset for this model, I can refer to the 3xM datasets - which have already the right format: https://github.com/xXAI-botXx/3xM
This refers to multiple datasets each with 20.000 syntethic datapoints.

I hope your request is solved by this answer. Have a good day!

---

DeepL translation:<br>
感谢您的提问。这个资源库只包含 Mask-RCNN 的代码，不包含数据集，所以我对您的问题的理解是："这个模型支持 COCO 格式的数据集吗？"这个模型支持 COCO 格式的数据集吗？
答案是：现在不支持，也没有计划支持。该模型的数据格式在[数据准备部分](https://github.com/xXAI-botXx/torch-mask-rcnn-instance-segmentation?tab=readme-ov-file#data-preparation)中有说明。

如果您需要该模型的数据集，我可以参考 3xM 数据集--它们已经有了正确的格式： https://github.com/xXAI-botXx/3xM
这指的是多个数据集，每个数据集有 20.000 个综合数据点。

希望您的问题能通过这个答案得到解决。祝您愉快

**Comment by 3120932798** [comment]
你好 想咨询你 这个数据集的标注原始文件可以开源吗 想把这个数据集转成其他模型支持的格式

**Comment by M-106** [comment]
Everything which I have is public available. There was never an extra annotation file. The ground truth masks are saved as gray images where each number is another object/instance -> 0 is the background, 1 is the first object, ... . I created the dataset in Unreal Engine 5.

You can download the wanted dataset from kaggle and then convert the gray image masks to a coco annotation file. That should be not difficult at all but the datasets are already usable with other models (depending on how you use them):

```python
from torch.utils.data import DataLoader
from maskrcnn_toolkit import Dual_Dir_Dataset, collate_fn  # you may want to copy the class and function from: https://github.com/xXAI-botXx/torch-mask-rcnn-instance-segmentation/blob/main/maskrcnn_toolkit.py

# DATA LOADING
data_mode = "random"
ground_path = "./path/to/dataset"  # FIXME
dataset_name = "3xM_Dataset_160_160"
dataset_name_2 = dataset_name.replace("Dataset_", "").lower().replace("_", "-")
dataset = Dual_Dir_Dataset(img_dir=os.path.join(ground_path, dataset_name_2, dataset_name, "rgb"), 
                           depth_dir=os.path.join(ground_path, dataset_name_2, dataset_name, "depth"), 
                           mask_dir=os.path.join(ground_path, dataset_name_2, dataset_name, "mask"), 
                           transform=None, 
                           amount=10, start_idx=0, end_idx=0, image_name="...", 
                           data_mode=data_mode, use_mask=True, use_depth=True, log_path="./logs",
                           width=1920, height=1080, should_log=True, should_print=True, should_verify=False)
data_loader = DataLoader(dataset, batch_size=5, shuffle=True, num_workers=4, collate_fn=collate_fn)

# MODEL
# load your model, optimizer, loss and scheduler here
# ...

# TRAINING
your_model.train()
for epoch in range(n_epochs):
    for images, targets, names in data_loader:
        images = list(image.to(device) for image in images)
        targets = [{k: v.to(device) for k, v in t.items()} for t in targets]

        loss_dict = your_model(images, targets)
        losses = sum(loss for loss in loss_dict.values())

        optimizer.zero_grad()    # gradient to zero
        losses.backward()        # create backward gradients
        optimizer.step()         # adjust weights towards gradients
        scheduler.step()         # adjust learnrate
```

---

Translation:
我所拥有的一切都是公开的。从来没有额外的注释文件。地面实况掩码保存为灰色图像，其中每个数字代表另一个对象/实例 -> 0 代表背景，1 代表第一个对象，......。我在虚幻引擎 5 中创建了数据集。

您可以从 kaggle 下载所需的数据集，然后将灰色图像掩码转换为 coco 注释文件。这一点都不难，但数据集已经可以用于其他模型（取决于您如何使用它们）：
*代码如下
